react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable prefer-const */
+import React, { useState } from 'react';
+
+export default function AppDrawer() {
+  const [iconClicked, setIconClicked] = useState(false);
+  const [image, setImage] = useState('https://i.imgur.com/7UWSnsD_d.webp?maxwidth=760&fidelity=grand');
+  const vaporeon = 'https://i.imgur.com/7UWSnsD_d.webp?maxwidth=760&fidelity=grand';
+  const gardevoir = 'https://i.imgur.com/qeAZRuq.png';
+  const lopunny = 'https://i.imgur.com/kerbwps.png';
+
+  const handleClick = (e) => {
+    setIconClicked(!iconClicked);
+    switch (e.currentTarget.id) {
+      case 'vaporeon':
+        setImage(vaporeon);
+        break;
+
+      case 'gardevoir':
+        setImage(gardevoir);
+        break;
+
+      case 'lopunny':
+        setImage(lopunny);
+        break;
+    }
+  };
+
+  let displayMenu = iconClicked;
+
+  return (
+    <>
+      <menu hidden={!displayMenu}>
+        <div>
+          <h3>Choose your Pokémon!</h3>
+          <li>
+            <button id='vaporeon' onClick={handleClick}>Vaporeon</button>
```

Comment:
I'm not sure you need the `<button>` since you can attach click handlers to `<li>` elements.